test/jdk/javax/security/auth/login/modules/SafeLogout.java

```diff
@@ -0,0 +1,111 @@
+/*
+ * Copyright (c) 2022, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+import javax.security.auth.Subject;
+import javax.security.auth.login.AppConfigurationEntry;
+import javax.security.auth.login.Configuration;
+import javax.security.auth.login.LoginContext;
+import javax.security.auth.login.LoginException;
+import java.util.*;
+
+import static javax.security.auth.login.AppConfigurationEntry.LoginModuleControlFlag.*;
+
+/*
+ * @test
+ * @bug 8282730
+ * @key randomness
+ * @summary LdapLoginModule throw NPE from logout method after login failure
+ * @modules jdk.security.auth
+ */
+public class SafeLogout {
+
+    static Random r = new Random();
+
+    public static void main(String[] args) throws Exception {
+        for (int i = 0; i < 100; i++) {
+            test(i);
+        }
+    }
+
+    static void test(int pos) throws Exception {
+        // Create random JAAS login config.
```

Comment:
I'm probably missing something obvious, but can you explain why this test uses a random login config? I would add some comments explaining that more.

Reply:
It's probably not necessary, I just wanted to understand your rationale. I would add a comment to the test like what you just explained above for future reference.